### PR TITLE
Implement AUTOLOAD_FILE constant

### PR DESCRIPTION
**Pull request by lpalgarvio**
This change implements AUTOLOAD_FILE constant to allow changing the autoload.php file being used by phpMyAdmin.

See obsolete pull request #13212

### CODE REVIEW

File: libraries/vendor_config.php [hunk 1]
```diff
@@ -14,6 +14,12 @@
 }
 
 /**
+ * Path to vendor autoload file. Useful when you want to
+ * have have vendor dependencies somewhere else.
+ */
+define('AUTOLOAD_FILE', './../vendor/autoload.php');
```
Comment (nijel) [inline libraries/vendor_config.php]: Is the path change intentional? Originally it was `./vendor/autoload.php` and now `./../vendor/autoload.php`.

Comment (lpalgarvio) [inline libraries/vendor_config.php]: Sorry, a mistake when I was doing tests

**Comment by nijel** [comment]
Ah, and the tests didn't fail because you don't use the new constant in tests bootstrap, please adjust it as well: https://github.com/phpmyadmin/phpmyadmin/blob/master/test/bootstrap-dist.php#L62

**Comment by lpalgarvio** [comment]
I was testing with the vendor directory and it's autoload.php outside phpmyadmin root, hence the reason it didn't failed.

There was a few more places mentioning autoload.php and i've fixed those as well.

PS: This project's require/include logic might be needing a major refactoring.

File: scripts/advisor2po [hunk 2]
```diff
@@ -50,8 +50,9 @@ function print_message($idx) {
 }
 
 define('PHPMYADMIN', 1);
-require_once 'vendor/autoload.php';
-require_once 'libraries/core.lib.php';
+require_once './../libraries/vendor_config.php';
+require_once './../' . AUTOLOAD_FILE;
+require_once './../libraries/core.lib.php';
```
Comment (nijel) [inline scripts/advisor2po]: This will break execution from root directory (see scripts/update-po).

File: scripts/generate-twig-cache [hunk 3]
```diff
@@ -1,8 +1,8 @@
 <?php
 /* vim: set expandtab sw=4 ts=4 sts=4 ft=php: */
 define('PHPMYADMIN', 1);
-require_once 'vendor/autoload.php';
-require_once 'libraries/vendor_config.php';
+require_once './../libraries/vendor_config.php';
+require_once './../' . AUTOLOAD_FILE;
```
Comment (nijel) [inline scripts/generate-twig-cache]: Same here, why do you change paths which work?

File: test/bootstrap-dist.php [hunk 4]
```diff
@@ -58,9 +58,9 @@
     }
 }
 
-require_once 'libraries/vendor_config.php';
-require_once 'vendor/autoload.php';
-require_once 'libraries/core.lib.php';
+require_once './../libraries/vendor_config.php';
+require_once './../' . AUTOLOAD_FILE;
+require_once './../libraries/core.lib.php';
```
Comment (nijel) [inline test/bootstrap-dist.php]: ...and here, see test failures on Travis.

File: scripts/generate-twig-cache (hunk 3 above)
Comment (lpalgarvio) [inline scripts/generate-twig-cache]: these looked as mistakes, as for example, in generate-twig-cache, these lines:
`$tplDir = dirname(__FILE__) . '/../templates';
$tmpDir = dirname(__FILE__) . '/../' . CACHE_DIR . 'twig';`
are assuming a different base path.
I also assumed that these scripts are being loaded directly via url or full path in the cli

Comment (nijel) [inline scripts/generate-twig-cache]: These use dirname(__FILE__)  as current dir, not the one from where script is executed.

Comment (lpalgarvio) [inline scripts/generate-twig-cache]: exactly, it's what I didn't gave enough attention.

I'm creating new commits

**Comment by nijel** [comment]
Squashed and merged, thanks for your contribution!

**Comment by lpalgarvio** [comment]
great, thanks =)

**Pull request merged**